myTeam.py

```diff
@@ -744,6 +810,26 @@ def offensive_heuristic(self, agent, goal, game_state, track_heuristics=False, e
 
         return heuristic
     
+    # Helper function that fixes the goal if it is not legal
+    def _fix_goal_if_not_legal(self, game_state):
+        # If any of the coordinates of the goal is negative, set that coordinate to 0
+        if self.goal[0] < 0:
+            self.goal = (0, self.goal[1])
+        if self.goal[1] < 0:
+            self.goal = (self.goal[0], 0)
+
+        # If any of the coordinates of the goal is greater than the width or height of the board, set that coordinate to the width or height of the board
+        if self.goal[0] > game_state.data.layout.width:
+            self.goal = (game_state.data.layout.width, self.goal[1])
+        if self.goal[1] > game_state.data.layout.height:
+            self.goal = (self.goal[0], game_state.data.layout.height)
+
```

Comment:
Same as previous comment
```suggestion
        if self.goal[0] >= game_state.data.layout.width:
            self.goal = (game_state.data.layout.width - 1, self.goal[1])
        if self.goal[1] >= game_state.data.layout.height:
            self.goal = (self.goal[0], game_state.data.layout.height - 1)

```